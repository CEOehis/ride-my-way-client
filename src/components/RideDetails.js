/* eslint-disable no-nested-ternary */
import React, { Fragment } from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import parseTime from '../utils/parseTime';

const RideDetails = ({
  ride,
  user,
  requested,
  handleRequest,
  requesting,
}) => {
  const {
    origin,
    destination,
    departureDate,
    departureTime,
    seats,
    rideId,
    rideCreator,
    email,
    phone,
  } = ride;
  if (!Object.keys(ride).length) return null;
  return (
    <Fragment>
      <h1>
        {origin}
        {' '}
        to
        {' '}
        {destination}
      </h1>
      <div className="ride-details">
        <div className="trip">
          <h1>Trip information</h1>
          <table>
            <tbody>
              <tr>
                <td>From: </td>
                <td>
                  {origin}
                </td>
              </tr>
              <tr>
                <td>To: </td>
                <td>
                  {destination}
                </td>
              </tr>
              <tr>
                <td>Departing: </td>
                <td>
                  {moment(departureDate).format('MMM DD, YYYY')}
                  ,
                  {parseTime(departureTime)}
                </td>
              </tr>
              <tr>
                <td>Available seats: </td>
                <td>
                  {seats}
                </td>
              </tr>
              <tr>
                <td>status: </td>
                <td>Available</td>
              </tr>
            </tbody>
          </table>
          <div className="call-to-action">
            {user.email !== email
              ? (
                <button
                  className={`btn btn-orange btn-lg ${requesting || requested ? 'submitting' : ''}`}
                  data-ride-id
                  href={rideId}
                  type="button"
                  disabled={requested}
                  onClick={handleRequest}
                >
                  {
                    requesting
                      ? 'requesting'
                      : requested ? 'request pending' : 'Join this Ride'
                  }
                </button>
              ) : ''
  }
            <Link className="btn btn-orange-inverse btn-lg" to="/home">More Rides</Link>
          </div>
        </div>
        <div className="driver-profile">
          <h1>Driver Profile</h1>
          <img src={`http://i.pravatar.cc/150?u=${email}`} alt="host-img" />
          <h2 className="driver-name">
            {rideCreator}
          </h2>
          <hr style={{ border: '3px solid #f1f1f1' }} />
          <p>
            email:
            {' '}
            {email}
          </p>
          <p>
            phone:
            {' '}
            {phone || '+234'}
          </p>
          <hr style={{ border: '3px solid #f1f1f1' }} />
        </div>
      </div>
    </Fragment>
  );
};

RideDetails.propTypes = {
  ride: PropTypes.object.isRequired,
  user: PropTypes.object.isRequired,
  requested: PropTypes.bool.isRequired,
  requesting: PropTypes.bool.isRequired,
  handleRequest: PropTypes.func.isRequired,
};

const mapStateToProps = ({ auth, rideDetails }) => ({
  user: auth.user,
  requested: rideDetails.requested,
  requesting: rideDetails.requesting,
});

export default connect(mapStateToProps)(RideDetails);
