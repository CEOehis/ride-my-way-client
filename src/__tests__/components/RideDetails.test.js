import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import RideDetails from '../../components/RideDetails';

Date.now = jest.fn(() => 1538380166917);
const state = {
  auth: {
    user: {
      email: 'jd@mail.com',
    },
  },
  rideDetails: {
    requested: false,
  },
};

const mockStore = configureMockStore();
const store = mockStore(state);

const ride = {
  createdAt: '2018-08-26T17:08:27.287Z',
  departureDate: Date.now(),
  departureTime: '12:30:00+01',
  destination: 'Port Liliane',
  origin: 'Reichelchester',
  rideCreator: 'Celestine Ekoh-Ordan',
  rideId: 1,
  seats: 2,
  updatedAt: '2018-08-26T17:08:27.287Z',
};

describe('RideDetails component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <RideDetails ride={ride} />
      </Provider>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
