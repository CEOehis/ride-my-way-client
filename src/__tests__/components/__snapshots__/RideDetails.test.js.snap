// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RideDetails component should render correctly 1`] = `
<Connect(RideDetails)
  ride={
    Object {
      "createdAt": "2018-08-26T17:08:27.287Z",
      "departureDate": 1538380166917,
      "departureTime": "12:30:00+01",
      "destination": "Port Liliane",
      "origin": "Reichelchester",
      "rideCreator": "Celestine Ekoh-Ordan",
      "rideId": 1,
      "seats": 2,
      "updatedAt": "2018-08-26T17:08:27.287Z",
    }
  }
/>
`;
